### PR TITLE
feat(core): add hover event dsl

### PR DESCRIPTION
**Pull request by LMLiam**
## Summary

Adds a Kotlin-first hover event DSL for Adventure text, item, and entity hover payloads, plus reusable hover-event factories and component/style scope application.

## Related Issues

Closes #22

## DSL Impact

- Adds top-level reusable `hover { ... }` events with `text(...)`, `item(...)`, and `entity(...)` payload selectors.
- Adds component/style scope support for `hover { ... }`, `hover(source)`, `hover(null)`, and typed `hover(action, value)` interop.
- Adds hover-event test matchers for component assertions.

## Rationale

Hover events are part of Adventure style state, so downstream Kotlin callers should be able to build and apply them with the same typed DSL shape as components, styles, and click events while still producing real Adventure `HoverEvent` payloads.

## Testing

- `./gradlew :test:test --tests io.github.lmliam.kotventure.test.text.HoverEventMatchersTest`
- `./gradlew :core:test --tests io.github.lmliam.kotventure.core.event.HoverEventDslTest`
- `./gradlew :serializer:test --tests io.github.lmliam.kotventure.serializer.ComponentSerializerExtensionsTest`
- `./gradlew ktlintFormat`
- `./gradlew ktlintCheck spotlessCheck`
- `./gradlew build`

## Checklist

- [x] Linked related issue
- [x] Updated README and relevant `/docs` pages (README and CHANGELOG updated; no separate docs page exists for this slice)
- [x] Verified examples build and run (DSL examples are covered by compiling unit tests with equivalent hover usage)

### CODE REVIEW

File: modules/core/src/main/kotlin/io/github/lmliam/kotventure/core/event/HoverContentScopeBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        check(!selected) {
+            "hover { ... } must choose only one payload: text(...), item(...), or entity(...)."
+        }
+        selected = true
+        this.event = event
+    }
+}
+
+private fun requireShowItemCount(count: Int): Int {
+    require(count >= 0) {
```
Comment (LMLiam) [inline modules/core/src/main/kotlin/io/github/lmliam/kotventure/core/event/HoverContentScopeBuilder.kt:75]: Review mechanics: The validation intentionally permits count = 0, but the tests only cover the negative rejection path. Please add a zero-count hover item test so the accepted boundary is locked down.

File: modules/core/src/main/kotlin/io/github/lmliam/kotventure/core/event/HoverContentScope.kt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        count: Int = 1,
+        dataComponents: Map<Key, DataComponentValue> = emptyMap(),
+    )
+
+    /**
+     * Selects an item hover payload from [item], [count], and typed [dataComponents].
+     *
+     * @throws IllegalArgumentException when [count] is negative.
+     */
+    public fun item(
```
Comment (LMLiam) [inline modules/core/src/main/kotlin/io/github/lmliam/kotventure/core/event/HoverContentScope.kt:51]: Review mechanics: These public Keyed overloads are part of the typed payload surface, but the DSL tests only exercise Key arguments. A regression in the Keyed forwarding path would not be caught; please add coverage for item/entity Keyed overloads.

Comment (LMLiam) [reply modules/core/src/main/kotlin/io/github/lmliam/kotventure/core/event/HoverContentScope.kt:51]: Fixed in 18397c4 by adding focused HoverEventDslTest coverage for both Keyed item and Keyed entity overloads.

File: modules/core/src/main/kotlin/io/github/lmliam/kotventure/core/event/HoverContentScopeBuilder.kt (hunk 1 above)
Comment (LMLiam) [reply modules/core/src/main/kotlin/io/github/lmliam/kotventure/core/event/HoverContentScopeBuilder.kt:75]: Fixed in 18397c4 by adding a zero-count item hover test, so the accepted boundary is covered alongside the negative-count rejection.

File: modules/core/src/test/kotlin/io/github/lmliam/kotventure/core/event/HoverEventDslTest.kt [hunk 3]
```diff
@@ -0,0 +1,224 @@
+package io.github.lmliam.kotventure.core.event
```
[the remark this answers is not in the document]
Comment (LMLiam) [reply modules/core/src/test/kotlin/io/github/lmliam/kotventure/core/event/HoverEventDslTest.kt:1]: Latest QDJVM/Qodana on head 18397c4 passed after the ktlint/Spotless/build verification, so this first-commit formatting note is no longer present on the updated PR head.

File: modules/test/src/test/kotlin/io/github/lmliam/kotventure/test/text/HoverEventMatchersTest.kt [hunk 4]
```diff
@@ -0,0 +1,188 @@
+package io.github.lmliam.kotventure.test.text
```
[the remark this answers is not in the document]
Comment (LMLiam) [reply modules/test/src/test/kotlin/io/github/lmliam/kotventure/test/text/HoverEventMatchersTest.kt:1]: Latest QDJVM/Qodana on head 18397c4 passed after the ktlint/Spotless/build verification, so this first-commit formatting note is no longer present on the updated PR head.

**Comment by LMLiam** [comment]
Review mechanics: CodeRabbit status check is red with `Insufficient usage credits` on head 18397c4. This is an external account/credits blocker rather than a code failure; I am not disabling or weakening the check. All other reported CI checks are passing.

File: modules/serializer/src/test/kotlin/io/github/lmliam/kotventure/serializer/ComponentSerializerExtensionsTest.kt [hunk 5]
```diff
@@ -117,5 +126,70 @@ class ComponentSerializerExtensionsTest :
...
+                val roundTripped = MiniMessage.miniMessage().deserialize(message.toMiniMessage())
+
+                roundTripped shouldHaveHoverText Component.text("Tooltip")
+            }
+
+            "round-trips item hover payload data components through MiniMessage" {
+                val dataComponents =
+                    mapOf<Key, DataComponentValue>(
+                        key("minecraft", "custom_data") to BinaryTagHolder.binaryTagHolder("{kotventure:1b}"),
+                    )
```
[the remark this answers is not in the document]
Comment (LMLiam) [reply modules/serializer/src/test/kotlin/io/github/lmliam/kotventure/serializer/ComponentSerializerExtensionsTest.kt:149]: Checked against the serializer module dependencies and Adventure 5.1.1 sources. `BinaryTagHolder` is the typed `DataComponentValue.TagSerializable` payload used here, and this test is scoped to MiniMessage hover-item payload preservation. Verifying SNBT parsing would require adding `adventure-nbt`/`TagStringIO` test coverage and would test NBT parser behavior rather than issue #22’s serializer round-trip acceptance criterion, so I am keeping this test scoped to the payload round trip.
